certbot-ci/certbot_integration_tests/utils/dns_server.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+"""Module to setup an RFC2136-capable DNS server"""
+import os
+import os.path
+from pkg_resources import resource_filename
+import shutil
+import socket
+import subprocess
+import sys
+import tempfile
+import time
+
+
+class DNSServer(object):
+    """
+    DNSServer configures and handles the lifetime of an RFC2136-capable server.
+    DNServer provides access to the dns_xdist parameter, listing the address and ports to
+    use for each pytest node.
+
+    At this time, DNSServer should only be used with a single node, but may be expanded in
+    future to support parallelization.
+    """
+
+    BIND_DOCKER_IMAGE = 'internetsystemsconsortium/bind9:9.16'
+    BIND_BIND_ADDRESS = ('127.0.0.1', 45953)
+
+    # A DNS message which is a query for '. IN A' transaction ID 0xe785. This is used by
+    # _wait_until_ready to check that BIND is responding without depending on dnspython.
+    BIND_TEST_QUERY = bytearray.fromhex('0028e7850120000100000000000100000100010000'
+                                        '29100000000000000c000a00083ad084e525a28702')
+
+
+    def __init__(self, nodes, show_output=False):
+        """
+        Create an DNSServer instance.
+        :param list nodes: list of node names that will be setup by pytest xdist
+        :param bool show_output: if True, print the output of the DNS server
+        """
+
+        self.bind_root = tempfile.mkdtemp()
+
+        # Same port to every node. We are only pretending to support parallelization.
```

Comment:
For the parallelization problem, I am more in favor of having one unique port and one unique DNS server, and configure in it dynamically the multiple zones required by all the workers, but this is more about a design discussion that we can have in a later PR.